J/Julia/build_tarballs.jl

```diff
@@ -1,103 +1,54 @@
-        BUILD_FLAGS+=("MARCH=x86-64" "JULIA_CPU_TARGET=generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)")
-    ;;
-    i686-*)
-        BUILD_FLAGS+=("MARCH=pentium4" "JULIA_CPU_TARGET=pentium4;sandybridge,-xsaveopt,clone_all")
-    ;;
-    arm-*)
-        BUILD_FLAGS+=("MARCH=armv7-a" "JULIA_CPU_TARGET=armv7-a;armv7-a,neon;armv7-a,neon,vfp4")
-    ;;
-    ppc64le-*)
-        BUILD_FLAGS+=("JULIA_CPU_TARGET=pwr8")
-    ;;
-    aarch64-*)
-        BUILD_FLAGS+=("MARCH=armv8-a" "JULIA_CPU_TARGET=generic")
-    ;;
-esac
-
-# If we're compiling for Windows, then set XC_HOST
-if [[ ${target} == *mingw* ]]; then
-    BUILD_FLAGS+=("XC_HOST=${target}")
-else
-    export LDFLAGS="$LDFLAGS -Wl,-rpath,${prefix}/lib"
-fi
-
-# Make use of many prebuilt things
-#for proj in BLAS LLVM PCRE MBEDTLS LIBSSH2 CURL LIBGIT2 GMP MPFR; do
-#    BUILD_FLAGS+=(USE_SYSTEM_${proj}=1)
-#done
-
-#if [[ ${nbits} == 64 ]]; then
-#    BUILD_FLAGS+=(LIBBLASNAME=libopenblas LIBBLAS=-lopenblas64_)
-#else
-#    BUILD_FLAGS+=(LIBBLASNAME=libopenblas LIBBLAS=-lopenblas)
-#fi
-
-#BUILD_FLAGS+=(LDFLAGS="$LDFLAGS")
-BUILD_FLAGS+=(LIBSSH2_ENABLE_TESTS=0)
-
-make ${BUILD_FLAGS[@]} -j${nproc} -C deps install-mbedtls
-make ${BUILD_FLAGS[@]} -j${nproc} -C deps install-libgit2
-make ${BUILD_FLAGS[@]} -j${nproc}
-make ${BUILD_FLAGS[@]} install
-"""
+version = v"1.4.1"
+verbose = "--verbose" in ARGS
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = [
-    Linux(:x86_64; libc=:glibc, compiler_abi=CompilerABI(:gcc7)),
-    Linux(:i686; libc=:glibc, compiler_abi=CompilerABI(:gcc7)),
-    Windows(:x86_64; compiler_abi=CompilerABI(:gcc7)),
-    Windows(:i686; compiler_abi=CompilerABI(:gcc7)),
+    Linux(:armv7l; libc=:glibc, compiler_abi=CompilerABI(libgfortran_version=v"4")),
+    Linux(:aarch64; libc=:glibc, compiler_abi=CompilerABI(libgfortran_version=v"4")),
+    Linux(:x86_64; libc=:glibc, compiler_abi=CompilerABI(libgfortran_version=v"4")),
+    Linux(:i686; libc=:glibc, compiler_abi=CompilerABI(libgfortran_version=v"4")),
+    MacOS(:x86_64; compiler_abi=CompilerABI(libgfortran_version=v"4")),
+    Windows(:x86_64; compiler_abi=CompilerABI(libgfortran_version=v"4")),
+    Windows(:i686; compiler_abi=CompilerABI(libgfortran_version=v"4")),
 ]
 
+# Function to map Julia platform/version to source url
+function tarball_url(p::Platform)
+    os_folder(p::Windows) = "winnt"
+    os_folder(p::MacOS) = "mac"
+    os_folder(p::Linux) = "linux"
+    os_folder(p::FreeBSD) = "freebsd"
+
+    os_tag(p::Windows) = "win$(wordsize(p))"
+    os_tag(p::MacOS) = "mac$(wordsize(p))"
+    function os_tag(p::Linux)
+        if arch(p) in (:armv7l, :aarch64, :ppc64le)
+            return "linux$(arch(p))"
+        end
+        return "linux$(wordsize(p))"
+    end
+    return "https://julialang2.s3.amazonaws.com/bin/$(os_folder(p))/x64/$(version.major).$(version.minor)/julia-$(version)-$(os_tag(p)).tar.gz"
+end
```

Comment:
Something like
```suggestion
function tarball_url(p::Platform)
    os_folder(p::Windows) = "winnt"
    os_folder(p::MacOS) = "mac"
    os_folder(p::Linux) = "linux"
    os_folder(p::FreeBSD) = "freebsd"

    arch_folder(p) = Dict{Symbol, String}(:i686    => "x86",
                                          :x86_64  => "x64",
                                          :aarch64 => "aarch64",
                                          :armv7l  => "armv7l",
                                          )[arch(p)]
    
    os_tag(p::Windows) = "win$(wordsize(p))"
    os_tag(p::MacOS) = "mac$(wordsize(p))"
    function os_tag(p::Linux)
        if arch(p) in (:armv7l, :aarch64, :ppc64le)
            return "linux-$(arch(p))"
        end
        return "linux-$(wordsize(p))"
    end
    return "https://julialang2.s3.amazonaws.com/bin/$(os_folder(p))/$(arch_folder(p))/$(version.major).$(version.minor)/julia-$(version)-$(os_tag(p)).tar.gz"
end
```
?  I don't know if there is an easier way to get the architecture folder

Reply:
I've been tweaking this code for too long.... I still have it stuck in my head that it's called `linux64.tar.gz`, not `linux-x86_64.tar.gz`